components/LinksHomepage/index.js

```diff
@@ -24,8 +24,10 @@ export default function LinksHomepage() {
     event.preventDefault();
 
     const newRecipe = { title, link };
-    setRecipes([...recipes, newRecipe]);
-    localStorage.setItem("recipes", JSON.stringify([...recipes, newRecipe]));
+    const updatedRecipes = [...recipes, newRecipe];
+
+    setRecipes(updatedRecipes);
+    localStorage.setItem("recipes", JSON.stringify(updatedRecipes));
```

Comment:
You should have a look into [use-local-storage-state](https://www.npmjs.com/package/use-local-storage-state) to handle syncing of state and and local storage. This is not necessary for this PR, but you could put it on your refactoring list: It initializes and synchronizes your state with local storage and avoids possible pitfalls with next.js server side rendering and hydration errors.

Reply:
thank you, Ralf I totally forgot about changing the API Key on Vercel. About use-local-storage-state, its already on my refactoring-todo-list for next week